examples/vision/mixUp.py

```diff
@@ -0,0 +1,226 @@
+x_train = np.reshape(x_train, (-1, 28, 28, 1))
+y_train = to_categorical(y_train, 10)
+
+x_test = x_test.astype("float32") / 255.0
+x_test = np.reshape(x_test, (-1, 28, 28, 1))
+y_test = to_categorical(y_test, 10)
+
+"""
+## Define hyperparameters
+"""
+
+AUTO = tf.data.AUTOTUNE
+BATCH_SIZE = 64
+EPOCHS = 5
+
+"""
+## Convert the data into TensorFlow `Dataset` objects
+"""
+
+train_ds_one = (
+    tf.data.Dataset.from_tensor_slices((x_train, y_train))
+    .shuffle(BATCH_SIZE * 100)
+    .batch(BATCH_SIZE)
+)
+train_ds_two = (
+    tf.data.Dataset.from_tensor_slices((x_train, y_train))
+    .shuffle(BATCH_SIZE * 100)
+    .batch(BATCH_SIZE)
+)
+# Because we will be mixing up the images and their corresponding labels, we will be
+# combining two shuffled datasets from the same training data.
+train_ds = tf.data.Dataset.zip((train_ds_one, train_ds_two))
+
+test_ds = tf.data.Dataset.from_tensor_slices((x_test, y_test)).batch(BATCH_SIZE)
+
+"""
+## Define the mixup technique function
+
+To perform the mixup routine, we create new virtual datasets using the training data from
+the same dataset, and apply a lambda value within the [0, 1] range sampled from a [Beta distribution](https://en.wikipedia.org/wiki/Beta_distribution)
+— such that, for example, `new_x = lambda * x1 + (1 - lambda) * x2` (where
+`x1` and `x2` are images) and the same equation is applied to the labels as well.
+"""
+
+
+def mix_up(ds_one, ds_two, alpha=0.2):
+    # Unpack two datasets
+    images_one, labels_one = ds_one
+    images_two, labels_two = ds_two
+    batch_size = tf.shape(images_one)[0]
+
+    # Sample lambda and reshape it to do the mixup
+    l = tf_distribution.Beta(0.2, 0.2).sample(batch_size)
+    x_l = tf.reshape(l, (batch_size, 1, 1, 1))
+    y_l = tf.reshape(l, (batch_size, 1))
+
+    # Perform mixup on both images and labels by combining a pair of images/labels
+    # (one from each dataset) into one image/label
+    images = images_one * x_l + images_two * (1 - x_l)
+    labels = labels_one * y_l + labels_two * (1 - y_l)
+    return (images, labels)
+
+
+"""
+**Note** that here , we are combining two images to create a single one. Theoretically,
+we can combine as many we want but that comes at an increased computation cost. In
+certain cases, it may not help improve the performance as well.
+"""
+
+"""
+## Visualize the new augmented dataset
+"""
+
+# First create the new dataset using our `mix_up` utility
+train_ds_mu = train_ds.map(
+    lambda ds_one, ds_two: mix_up(ds_one, ds_two, alpha=0.2), num_parallel_calls=AUTO
+)
+
+# Let's preview 9 samples from the dataset
+sample_images, sample_labels = next(iter(train_ds_mu))
+plt.figure(figsize=(10, 10))
+for i, (image, label) in enumerate(zip(sample_images[:9], sample_labels[:9])):
+    ax = plt.subplot(3, 3, i + 1)
+    plt.imshow(image.numpy().squeeze())
+    print(label.numpy().tolist())
+    plt.axis("off")
+
+"""
+## Model building
+"""
+
+
+def get_training_model():
+    model = tf.keras.Sequential(
```

Comment:
You can define the model by passing a list of layers to the constructor, which is shorter.